src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx

```diff
@@ -21,91 +21,86 @@ import TextBlock from 'app/views/settings/components/text/textBlock';
+  return (
+    <React.Fragment>
+      <DateLabel>{label}</DateLabel>
+      <div>{date ? <DateTime date={date} /> : t('never')}</div>
+    </React.Fragment>
+  );
 }
 
-class AccountSecurityDetails extends AsyncView {
-  static PropTypes = {
-    deleteDisabled: PropTypes.bool.isRequired,
-    onRegenerateBackupCodes: PropTypes.func.isRequired,
-  };
+type Props = {
+  deleteDisabled: boolean;
+  onRegenerateBackupCodes: () => void;
+} & RouteComponentProps<{authId: string}, {}>;
 
-  _form = {};
+type State = {
+  authenticator: Authenticator | null;
+} & AsyncView['state'];
 
+class AccountSecurityDetails extends AsyncView<Props, State> {
   getTitle() {
     return t('Security');
   }
 
-  getEndpoints() {
-    return [['authenticator', `${ENDPOINT}${this.props.params.authId}/`]];
-  }
+  getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
+    const {params} = this.props;
+    const {authId} = params;
 
-  addError(message) {
-    this.setState({loading: false});
-    addErrorMessage(message);
+    return [['authenticator', `${ENDPOINT}${authId}/`]];
   }
 
-  handleRemove = device => {
+  handleRemove = async (device?: Authenticator['devices'][0]) => {
     const {authenticator} = this.state;
 
     if (!authenticator || !authenticator.authId) {
       return;
     }
-    const isRemovingU2fDevice = !!device;
-    const deviceId = isRemovingU2fDevice ? `${device.key_handle}/` : '';
-
-    this.setState(
-      {
-        loading: true,
-      },
-      () =>
-        this.api
-          .requestPromise(`${ENDPOINT}${authenticator.authId}/${deviceId}`, {
-            method: 'DELETE',
-          })
-          .then(
-            () => {
-              this.props.router.push('/settings/account/security');
-              const deviceName = isRemovingU2fDevice ? device.name : 'Authenticator';
-              addSuccessMessage(t('%s has been removed', deviceName));
-            },
-            () => {
-              // Error deleting authenticator
-              const deviceName = isRemovingU2fDevice ? device.name : 'authenticator';
-              this.addError(t('Error removing %s', deviceName));
-            }
-          )
-    );
+
+    // if the device is defined, it means that U2f is being removed
+    const deviceId = device ? `${device.key_handle}/` : '';
+    const deviceName = device ? device.name : t('Authenticator');
+
+    this.setState({loading: true});
+
+    try {
+      await this.api.requestPromise(`${ENDPOINT}${authenticator.authId}/${deviceId}`, {
+        method: 'DELETE',
+      });
+      this.props.router.push('/settings/account/security');
+      addSuccessMessage(t('%s has been removed', deviceName));
+    } catch {
+      // Error deleting authenticator
+      this.setState({loading: false});
+      addErrorMessage(t('Error removing %s', deviceName));
+    }
   };
 
   renderBody() {
     const {authenticator} = this.state;
+
+    if (!authenticator) {
```

Comment:
This is what i meant about adding `null` to `AsyncComponent` things, `renderBody` will never render when state from requests is null

Reply:
that's true, but I prefer to leave it at that for now...let's discuss this topic in the upcoming TSC meeting 😉